src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,274 @@
+        if (userInput.length() <= command.length()) {
+            throw new DukeException(exceptionMessage);
+        }
+
+        // If userInput after command is not space, tell users that they are missing space.
+        if (userInput.charAt(command.length()) != ' ') {
+            throw new DukeException(Ui.exceptionMissingSpaceAfterCommand(command));
+        } else {
+            // If user input is like so "command ", it also does not contain desired inputs, despite having space.
+            if (userInput.length() == (command.length() + 1)) {
+                throw new DukeException(exceptionMessage);
+            }
+        }
+    }
+
+    /**
+     * Prepends zeroes to the string till a desired length of the output string.
+     *
+     * @param original String to prepend zeroes to.
+     * @param expected Length of output string.
+     * @return String prepended with zeroes till desired length.
+     * @throws DukeException If string input is larger than expected length of output string.
+     */
+    private static String padZeros(String original, int expected) throws DukeException {
+        String output = original;
+        if (original.length() < expected) {
+            for (int i = 0; i < (expected - original.length()); i++) {
+                output = "0" + original;
+            }
+        } else if (original.length() > expected) {
+            throw new DukeException(Ui.exceptionInvalidDateTimeFormat());
+        }
+
+        return output;
+    }
+
+    /**
+     * Parses a LocalDate into a string format "d MMMM yyyy" to be printed to the user.
+     * For instance, the LocalDate equivalent of 2020-01-01 will be parsed into "1 January 2020".
+     *
+     * @param localDate LocalDate to be parsed.
+     * @return Formatted string representing the date of LocalDate in d MMMM yyyy format.
+     * @throws DukeException
+     */
+    public static String parseLocalDate(LocalDate localDate) throws DukeException {
+        try {
+            return localDate.format(DateTimeFormatter.ofPattern("d MMMM yyyy"));
+        } catch (DateTimeException dateTimeException) {
+            throw new DukeException(Ui.exceptionInvalidLocalDate());
+        }
+    }
+
+    /**
+     * Parses a formatted string into LocalDate.
+     * The formatted string must be of the format "dd/mm/yyyy".
+     *
+     * @param dateString Formatted string to be parsed.
+     * @return LocalDate equivalent of the date represented by dateString.
+     * @throws DukeException If dateString is of an invalid date format.
+     * @throws DukeException If dateString represents a date later than "+999999999-12-31" or
+     *                       earlier than "-999999999-01-01".
+     */
+    public static LocalDate toLocalDate(String dateString) throws DukeException {
+        String[] split = dateString.split("[/\\s]");
+
+        // Check if split has 3 elements. If not, this is already an invalid date format.
```

Comment:
Perhaps a comment here might no be as necessary? I think the code is simple enough!

Reply:
I agree that the code is simple, but I would still prefer to keep the comment for clarity purposes.